app/guid-file/controller.ts

```diff
@@ -180,7 +193,131 @@ export default class GuidFile extends Controller {
 
     @action
     versionChange(version: number) {
-        this.set('revision', +version);
+        this.set('revision', version);
+    }
+
+    @action
+    async toggleVersions() {
+        this.toggleProperty('revisionClicked');
+
+        if (this.revisionClicked === true) {
+            this.openVersions();
+        }
+
+        if (this.tagsClicked === true) {
+            this.closeTags();
+        }
+
+        if (this.revisionClicked === false) {
+            this.closeVersions();
+        }
+    }
+
+    @action
+    async toggleTags() {
+        this.toggleProperty('tagsClicked');
+
+        if (this.tagsClicked === true) {
+            this.openTags();
+        }
+        if (this.revisionClicked === true) {
+            this.closeVersions();
+        }
+        if (this.tagsClicked === false) {
+            this.closeTags();
+        }
+    }
+
+    @action
+    async showMoreButtons() {
+        console.log('Open more button clicked');
+    }
+
+    @action
+    async share() {
+        console.log('File shared with service:');
+    }
+
+    openVersions() {
+
+        if (this.tagsClicked === true) {
+            this.closeTags();
+            this.tagsClicked = false;
+        }
+
+        const rightPanel = document.getElementById('rightPanel');
+        if (rightPanel) {
+            rightPanel.style.marginLeft = '-400px';
+            rightPanel.classList.replace('col-sm-1', 'col-sm-5');
```

Comment:
In general, we prefer to change the class of the item rather than adjusting the styles programatically. The new class can have the different styles. That being said, it might also be worthwhile using Bootstrap Tabs rather than doing this manually, presuming we can also layer in the behavior to hide the whole panel if you click on an open tab.

Don't change this for the moment, because we can do that in the versions or tags ticket.

Reply:
Ok, that sounds good. I had worked previously with Bootstrap Nav and Tabs so we can also test that out along with replacing the class. This had been less code and had been programmatically interesting, but we can implement whichever works best.